azure-pipelines.yaml

```diff
@@ -0,0 +1,86 @@
+# Starter pipeline
+# Build a general Node.js project with npm.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+
+name: $(Date:yyyyMMdd)$(Rev:.r)
+
+trigger:
+  - main
+  - features/*
+  - feature/*
+
+variables:
+  isMaster: $[eq(variables['Build.SourceBranch'], 'refs/heads/master')]
+  isPullRequest: $[eq(variables['Build.Reason'], 'PullRequest')]
+  branchSource: $[replace(variables['Build.SourceBranch'], 'refs/heads/', '')]
+  branch: $[replace(variables['branchSource'], '/', '-')]
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+steps:
+  - task: UseNode@1
+    inputs:
+      version: '18.x'
+    displayName: 'Install Node.js'
+
+  - task: SonarCloudPrepare@1
+    inputs:
+      SonarCloud: 'defra-sonarcloud'
+      organization: 'defra'
+      scannerMode: CLI
+      projectKey: 'DEFRA_ncea-frontend'
+      projectName: 'ncea-frontend'
+    displayName: 'Prepare analysis on SonarCloud'
+
+  - script: |
+      npm install
+      npm run build
+    displayName: 'npm install and build'
+
+  - script: |
+      npm run test
+    displayName: 'npm test'
+
+  - task: PublishCodeCoverageResults@1
+    condition: succeededOrFailed()
+    inputs:
+      codeCoverageTool: Cobertura
+      summaryFileLocation: $(System.DefaultWorkingDirectory)/coverage/cobertura-coverage.xml
+    displayName: 'Publish code coverage result'
+
+  - task: SonarCloudAnalyze@1
+    inputs:
+      jdkversion: 'JAVA_HOME_17_X64'
+    displayName: 'Run Code Analysis'
+
+  - task: SonarCloudPublish@1
+    inputs:
+      pollingTimeoutSec: '300'
+    displayName: 'Publish Quality Gate Result'
+
+  - task: CopyFiles@2
+    inputs:
+      sourceFolder: '$(Build.SourcesDirectory)'
+      contents: |
+        build/**
+      targetFolder: '$(Build.ArtifactStagingDirectory)'
+      cleanTargetFolder: true
+    displayName: 'Copy project files'
+
+  - task: PublishPipelineArtifact@1
+    condition: and(and(succeeded(), ne(variables.isMaster, true)), ne(variables.isPullRequest, true))
+    inputs:
+      artifactName: 'ncea-frontend_$(Build.BuildNumber)_$(branch)'
+      targetPath: '$(Build.ArtifactStagingDirectory)'
+      publishLocation: 'pipeline'
+    displayName: 'Publish npm artifact'
+
+  - task: PublishPipelineArtifact@1
```

Comment:
- Instead of having two artifact task we have a condition for articaft publishing
- We can consolidate the condition for both `CopyFiles` and `PublishPipelineArtifact`
eg:
<img width="458" alt="image" src="https://github.com/DEFRA/ncea-frontend/assets/131792244/67b4c7ff-729d-4d1e-a1db-beae45c5bc6c">

Reply:
Remove the duplicate artifact task